All major changes of version 9.0 by @bobsik

GoAlexander opened the pull request:
ADDED::
- Encoding manipulations
- opening different files than utf-8 should work better, you can specify your region or direct custom encoding type (by Python standards) (Asia cooperation needed, what are best encodings you use in Korea,China, Japan etc.)
- possibility to open file with different encoding
- possibility to save file with different encoding

CHANGED:
- Re-creation of the UI (a bit more informative and more minimized)
- On Ambience Theme you can pick your own color for the UI (buttons and some labels, but not all)
- better Ambience of the OS manipulation, mainly resolving differences on change between light or dark type of ambience
- few icons changed, button's icons have the same color as button's text
- "copy file path" - when you extend the menu on right side you have the path to file and encoding, just tap there to copy

FIXED:
- Line count works much more better(!), recounting on changing Landscape/Portrait mode

Known bugs:
- Combobox of tab can not be dynamically changed. It needs to reopen SettingsPage to see the changes.

TODO:
- [x] Check possible errors with `settings` and `history` config files in transition(!) of 0.8.x->0.9 
- [x] Check user friendly interface
- [x] Update contributors list
- [x] Update changelog
- [x] Update version on AboutPage

GoAlexander commented:
@bobsik I tested version 0.9, found bugs:  
1) "Quick note" button in the app does not work - it does not open  
2) "R-only" button is not bright if enabled  
3) Removed button "copy file path to the clipboard"  

No bugs on line counting found! I think that it needs to fix bugs above and I can release version 0.9 :)

Bobsik commented:
Hello,

1. I had not actually used it, so I have no idea now :)
2. I forgot to edit this, like you have probably noticed I have changed 
color behaviour, it just need one line of code
3. it is not removed, but actually only moved, when you extend the menu on 
right side you have the path to file and encoding, just tap there :) (this 
would be probably nice to comment  in descripiton since it is not obviously 
obvious)

I will look at #1 and just add #2.
-- 
Best regards
Petr Skoumal

---------- Původní e-mail ----------
Od: Alexander Dydychkin <notifications@github.com>
Komu: GoAlexander/harbour-editor <harbour-editor@noreply.github.com>
Datum: 22. 3. 2020 21:43:30
Předmět: Re: [GoAlexander/harbour-editor] All major changes of version 9.0 
by @bobsik (#114) 
"


@Bobsik(https://github.com/Bobsik) I tested version 0.9, found bugs:

   1. "Quick note" button in the app does not work - it does not open
   2. "R-only" button is not bright if enabled
   3. Removed button "copy file path to the clipboard"
   

No bugs on line counting found! I think that it needs to fix bugs above and 
I can release version 0.9 :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
(https://github.com/GoAlexander/harbour-editor/pull/114#issuecomment-602270317)
, or unsubscribe
(https://github.com/notifications/unsubscribe-auth/AJMGKGYCBL53XLMHOPINGWTRIZZ6ZANCNFSM4LRH4JUQ)
. 
"

Bobsik commented:
Ok, even #1 is now clickable, but it does not save, I basically know why, 
but it will take little bit more time then few clicks.. Will let you know :)

-- 
Petr Skoumal

---------- Původní e-mail ----------
Od: petr.skoum@seznam.cz
Komu: GoAlexander/harbour-editor <reply+AJMGKG24W7SMB55PUTZSBMN4
QOXOZEVBNHHCF2HQVQ@reply.github.com>
Datum: 24. 3. 2020 13:00:47
Předmět: Re: [GoAlexander/harbour-editor] All major changes of version 9.0 
by @bobsik (#114) 
"Hello,

1. I had not actually used it, so I have no idea now :)
2. I forgot to edit this, like you have probably noticed I have changed 
color behaviour, it just need one line of code
3. it is not removed, but actually only moved, when you extend the menu on 
right side you have the path to file and encoding, just tap there :) (this 
would be probably nice to comment  in descripiton since it is not obviously 
obvious)

I will look at #1 and just add #2.
-- 
Best regards
Petr Skoumal

---------- Původní e-mail ----------
Od: Alexander Dydychkin <notifications@github.com>
Komu: GoAlexander/harbour-editor <harbour-editor@noreply.github.com>
Datum: 22. 3. 2020 21:43:30
Předmět: Re: [GoAlexander/harbour-editor] All major changes of version 9.0 
by @bobsik (#114) 
"


@Bobsik(https://github.com/Bobsik) I tested version 0.9, found bugs:

   1. "Quick note" button in the app does not work - it does not open
   2. "R-only" button is not bright if enabled
   3. Removed button "copy file path to the clipboard"
   

No bugs on line counting found! I think that it needs to fix bugs above and 
I can release version 0.9 :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
(https://github.com/GoAlexander/harbour-editor/pull/114#issuecomment-602270317)
, or unsubscribe
(https://github.com/notifications/unsubscribe-auth/AJMGKGYCBL53XLMHOPINGWTRIZZ6ZANCNFSM4LRH4JUQ)
. 
""

Bobsik commented:
Done, Git updated :)

-- 
Petr Skoumal

---------- Původní e-mail ----------
Od: petr.skoum@seznam.cz
Komu: GoAlexander/harbour-editor <reply+AJMGKG24W7SMB55PUTZSBMN4
QOXOZEVBNHHCF2HQVQ@reply.github.com>
Datum: 24. 3. 2020 13:10:55
Předmět: Re: [GoAlexander/harbour-editor] All major changes of version 9.0 
by @bobsik (#114) 
"Ok, even #1 is now clickable, but it does not save, I basically know why, 
but it will take little bit more time then few clicks.. Will let you know :)

-- 
Petr Skoumal

---------- Původní e-mail ----------
Od: petr.skoum@seznam.cz
Komu: GoAlexander/harbour-editor <reply+AJMGKG24W7SMB55PUTZSBMN4
QOXOZEVBNHHCF2HQVQ@reply.github.com>
Datum: 24. 3. 2020 13:00:47
Předmět: Re: [GoAlexander/harbour-editor] All major changes of version 9.0 
by @bobsik (#114) 
"Hello,

1. I had not actually used it, so I have no idea now :)
2. I forgot to edit this, like you have probably noticed I have changed 
color behaviour, it just need one line of code
3. it is not removed, but actually only moved, when you extend the menu on 
right side you have the path to file and encoding, just tap there :) (this 
would be probably nice to comment  in descripiton since it is not obviously 
obvious)

I will look at #1 and just add #2.
-- 
Best regards
Petr Skoumal

---------- Původní e-mail ----------
Od: Alexander Dydychkin <notifications@github.com>
Komu: GoAlexander/harbour-editor <harbour-editor@noreply.github.com>
Datum: 22. 3. 2020 21:43:30
Předmět: Re: [GoAlexander/harbour-editor] All major changes of version 9.0 
by @bobsik (#114) 
"


@Bobsik(https://github.com/Bobsik) I tested version 0.9, found bugs:

   1. "Quick note" button in the app does not work - it does not open
   2. "R-only" button is not bright if enabled
   3. Removed button "copy file path to the clipboard"
   

No bugs on line counting found! I think that it needs to fix bugs above and 
I can release version 0.9 :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
(https://github.com/GoAlexander/harbour-editor/pull/114#issuecomment-602270317)
, or unsubscribe
(https://github.com/notifications/unsubscribe-auth/AJMGKGYCBL53XLMHOPINGWTRIZZ6ZANCNFSM4LRH4JUQ)
. 
"""

GoAlexander commented:
Where updated Git?

GoAlexander commented:
Hi,
Did not know that you use emails for answers, I updated my previous comment:
@Bobsik I tested version 0.9, found bugs:

1. "Quick note" button in the app does not work - it does not open **- does it fixed?**
2. "R-only" button is not bright if enabled **- understood FIXED**
3. Removed button "copy file path to the clipboard" **- understood**
4. Removed button "open quicknote" on CoverPage. Why? :)
5. Need background for your new menu if Ambience Theme enabled (if text is under the buttons they are unreadable)
6. Found the bug connected with line counting if font size changed: https://www.dropbox.com/s/v7uk8ccu1eq93xg/Screenshot_20200322_004.png

I think that it needs to fix bugs above plus TODO list above and I can release version 0. :)

Bobsik commented:
Hello,

no, Git is not my friend, only data store :)
1-3 yes
4 first time I have tried the app there were some reports in cml with errors
regarding coverpage, I have commented it in code aaand then forgot about 
that. Will look.
5 probably did not notice, will do
6 will look, I have not tried another fontsizes after that

-- 
Petr Skoumal

---------- Původní e-mail ----------
Od: Alexander Dydychkin <notifications@github.com>
Komu: GoAlexander/harbour-editor <harbour-editor@noreply.github.com>
Datum: 6. 4. 2020 0:16:10
Předmět: Re: [GoAlexander/harbour-editor] All major changes of version 9.0 
by @bobsik (#114) 
"


Hi,
Did not know that you use emails for answers, I updated my previous comment:
@Bobsik(https://github.com/Bobsik) I tested version 0.9, found bugs:

   1. "Quick note" button in the app does not work - it does not open - does
   it fixed?
   2. "R-only" button is not bright if enabled - understood FIXED
   3. Removed button "copy file path to the clipboard" - understood
   4. Removed button "open quicknote" on CoverPage. Why? :)
   5. Need background for your new menu if Ambience Theme enabled (if text 
   is under the buttons they are unreadable)
   6. Found the bug connected with line counting if font size changed: 
   https://www.dropbox.com/s/v7uk8ccu1eq93xg/Screenshot_20200322_004.png
   (https://www.dropbox.com/s/v7uk8ccu1eq93xg/Screenshot_20200322_004.png)
   

I think that it needs to fix bugs above plus TODO list above and I can 
release version 0. :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
(https://github.com/GoAlexander/harbour-editor/pull/114#issuecomment-609493338)
, or unsubscribe
(https://github.com/notifications/unsubscribe-auth/AJMGKGZJCJWBHFFWNWRKSALRLD7FLANCNFSM4LRH4JUQ)
. 
"

GoAlexander commented:
Thanks :)

GoAlexander commented:
"Done, Git updated :)"  
  
Hm, where you pushed your commits? Do not see any changes... Can you share the link with branch with your commits?

Bobsik commented:
Hello, 

Like I said, git is not my friend, I am literally using it on the end, when 
I feel I have finished, I won't update any tiny stuff, it is unnecessary 
administrative work.. If I would have worked with more people, it might be 
useful to do it more frequent, but this way it is completely pointless. 

anyway I have pushed (overwrite, because that stupid git did not want to do 
it without it) all the files. 

I have repaired what we have been talked about, and also added better 
manipulation with font size and type in the settings, so it can be better 
visible actual font. I wanted to have font type specified directly in 
ComboBox, but Sailfish seems to probably block it for the sake of the 
"consistent GUI".. So I had to do it with additional "text testing" 
textfield.

Petr

-- 
Děkuji a s pozdravem
Petr Skoumal

---------- Původní e-mail ----------
Od: Alexander Dydychkin <notifications@github.com>
Komu: GoAlexander/harbour-editor <harbour-editor@noreply.github.com>
Datum: 9. 4. 2020 11:22:18
Předmět: Re: [GoAlexander/harbour-editor] All major changes of version 9.0 
by @bobsik (#114) 
"


"Done, Git updated :)"

Hm, where you pushed your commits? Do not see any changes... Can you share 
the link with branch with your commits?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
(https://github.com/GoAlexander/harbour-editor/pull/114#issuecomment-611426530)
, or unsubscribe
(https://github.com/notifications/unsubscribe-auth/AJMGKG667H4FPV63ENVHGWDRLWHTHANCNFSM4LRH4JUQ)
. 
"

GoAlexander commented:
Ok, understood. Will try to publish at the weekend.  
Is it the latest version https://github.com/Bobsik/harbour-editor/commit/d8dc76662198d7692599ccf68e30af5edace7032 , yeah?

GoAlexander commented:
Uploaded changes from https://github.com/Bobsik/harbour-editor/commit/d8dc76662198d7692599ccf68e30af5edace7032 to this branch

GoAlexander commented:
@Bobsik I have some troubles with my SaifishOS build emulator. Strange but it is damaged and cannot be started. So can you please:
- download this branch `updates_of_bobsik` (I pushed small change for version and AboutPage)
- build release version for arm and x86 platform
- send them per email or here  
  
I wanted to test and publish but cannot to build latest version :(

GoAlexander commented:
@Bobsik Ok, I fixed my problem with emulator. I see one bug with text position on CoverPage, will try to fix it by my own and will publish it.
